sdk/rust/src/http/router.rs

```diff
@@ -1,17 +1,51 @@
+// This router implementation is heavily inspired by the `Endpoint` type in the https://github.com/http-rs/tide project.
+
 use super::conversions::{IntoResponse, TryFromRequest, TryIntoRequest};
 use super::{responses, Method, Request, Response};
+use async_trait::async_trait;
 use routefinder::{Captures, Router as MethodRouter};
+use std::future::Future;
 use std::{collections::HashMap, fmt::Display};
 
-type Handler = dyn Fn(Request, Params) -> Response;
+type DynHandler = dyn Handler;
```

Comment:
I'd prefer not to have type aliases unless they are obvious and make things much easier to read which I don't think is the case here.